src/components/anchor-navigation/anchor-navigation.test.tsx

```diff
@@ -0,0 +1,364 @@
+}
+const Content = React.forwardRef<HTMLDivElement, ContentProps>(
+  (
+    { title, noTextbox, tabIndex, "data-role": dataRole }: ContentProps,
+    ref
+  ) => (
+    <>
+      <div
+        ref={ref}
+        className="focusableContent"
+        tabIndex={tabIndex}
+        data-role={dataRole}
+      >
+        {!noTextbox && <Textbox label={title} value="" onChange={() => {}} />}
+        <h2>{title}</h2>
+      </div>
+    </>
+  )
+);
+
+Element.prototype.scrollIntoView = jest
+  .fn()
+  .mockImplementation(function mockView(
+    this: HTMLDivElement,
+    options: ScrollIntoViewOptions
+  ) {
+    return { element: this, options };
+  });
+
+let ref1: React.RefObject<HTMLDivElement>;
+let ref2: React.RefObject<HTMLDivElement>;
+let ref3: React.RefObject<HTMLDivElement>;
+let ref4: React.RefObject<HTMLDivElement>;
+let ref5: React.RefObject<HTMLDivElement>;
+
+const renderComponent = (props?: AnchorNavigationProps) => {
+  ref1 = React.createRef();
+  ref2 = React.createRef();
+  ref3 = React.createRef();
+  ref4 = React.createRef();
+  ref5 = React.createRef();
+
+  return render(
+    <AnchorNavigation
+      stickyNavigation={
+        <>
+          <AnchorNavigationItem target={ref1}>First</AnchorNavigationItem>
+          <AnchorNavigationItem target={ref2}>Second</AnchorNavigationItem>
+          <AnchorNavigationItem target={ref3}>Third</AnchorNavigationItem>
+          <AnchorNavigationItem target={ref4}>
+            The slightly longer than expected fourth navigation item
+          </AnchorNavigationItem>
+          <AnchorNavigationItem target={ref5}>Fifth</AnchorNavigationItem>
+        </>
+      }
+      {...props}
+    >
+      <Content
+        tabIndex={0}
+        ref={ref1}
+        title="First section"
+        data-role="section-1"
+      />
+      <AnchorSectionDivider />
+      <Content ref={ref2} title="Second section" data-role="section-2" />
+      <AnchorSectionDivider />
+      <Content
+        ref={ref3}
+        title="Third section"
+        noTextbox
+        data-role="section-3"
+      />
+      <AnchorSectionDivider />
+      <Content ref={ref4} title="Fourth section" data-role="section-4" />
+      <AnchorSectionDivider />
+      <Content ref={ref5} title="Fifth section" data-role="section-5" />
+    </AnchorNavigation>
+  );
+};
+
+beforeEach(() => {
+  jest.useFakeTimers();
+});
+
+afterEach(() => {
+  jest.useRealTimers();
+});
+
+test("has proper data attributes applied to elements", () => {
+  renderComponent({ "data-role": "tags-test" });
+  expect(screen.getByTestId("tags-test")).toHaveAttribute(
+    "data-component",
+    "anchor-navigation"
+  );
+  expect(screen.getByRole("list")).toHaveAttribute(
+    "data-element",
+    "anchor-sticky-navigation"
+  );
+
+  const anchorNavigationLinks = screen.getAllByRole("link");
+  anchorNavigationLinks.forEach((anchor) => {
+    expect(anchor).toHaveAttribute("data-element", "anchor-navigation-item");
+  });
+});
+
+test.each([0, 1, 2, 3, 4])(
```

Comment:
it's a fair question. I did it this way mainly because that's how we were doing it in the Enzyme tests, and I'd always rather have too many tests than too few if unsure. But I think you're right - I'm happy to change it to just testing one (not the first one as that's pre-selected so the test might still pass even if this got broken).